src/main/java/decodex/parser/Parser.java

```diff
@@ -1,5 +1,75 @@
 package decodex.parser;
 
+import decodex.commands.Command;
+import decodex.commands.ExitCommand;
+import decodex.data.DataManager;
+import decodex.modules.ModuleManager;
+import decodex.parser.exception.ParserException;
+import decodex.ui.Ui;
+
+import java.util.Arrays;
+
+/**
+ * Parses and validates the user input.
+ */
 public class Parser {
 
+    /**
+     * Returns the type of command that user has entered.
+     *
+     * @param userInput The input specified by the user.
+     * @return The type of command.
+     */
+    public String getCommandType(String userInput) throws ParserException {
+        String[] tokens = userInput.split(" ");
+        String commandType = tokens[0];
+        if (commandType.isEmpty()) {
+            throw new ParserException(ParserException.MISSING_COMMAND_TYPE_MESSAGE);
+        }
+        return tokens[0];
+    }
+
+    /**
+     * Returns the Argument portion of the user input.
+     *
+     * @param userInput The input specified by the user.
+     * @return The argument portion of the user input.
+     */
+    public String[] getUserArguments(String userInput) throws ParserException {
+        try {
+            String[] tokens = userInput.split(" ");
+            return Arrays.copyOfRange(tokens, 1, tokens.length);
+        } catch (ArrayIndexOutOfBoundsException err) {
+            throw new ParserException(ParserException.MISSING_COMMAND_ARGS_MESSAGE);
+        }

```

Comment:
Not advisable to re-throw another exception. Because the `ArrayIndexOutOfBoundsException` could be caused when accessing the element at index `1` during the `copyOfRange` call, perhaps throw the `ParserException` when the size of `tokens` is `<= 1` ?